src/components/editor/ImportResourceTemplate.jsx

```
@@ -43,80 +44,102 @@ class ImportResourceTemplate extends Component {
-        this.state.updateResourceError ? this.setState({message: [`${msg}: ${this.state.updateResourceError}`]}) : this.setState({message: [msg]})
-      }
-    })
+    console.log('received non-error response to create resource request:')
+    console.dir(response)
+    return response.response
   }
 
-  updateStateForResourceError = (error) => {
-    if(_.get(error, 'response.statusText') === 'Conflict') {
-      const response = error.response
-      const joinedConflicts = this.state.createResourceError.slice(0)
-      this.setState({tempConflictError: response})
-      joinedConflicts.push(this.state.tempConflictError)
-      this.setState({createResourceError: joinedConflicts})
+  updateResource = async (content, group) => {
+    let response
+    try {
+      response = await instance.updateResourceWithHttpInfo(group, content.id, content, { contentType: 'application/json' })
+    } catch(error) {
+      console.log('received error with update resource request:')
+      console.dir(error)
+      this.setState({ updateResourceError: error })
+      return error.response
     }
+    console.log('received non-error response to update resource request:')
+    console.dir(response)
+    return response.response
   }
 
-  updateAllResources = (rts, group) => {
-    return Promise.all(rts.map(async rt =>
-      await this.fulfillCreateResourcePromise(this.updateResourcePromise(rt, group))
-    ))
+  handleResponse = (response, operation, profileCount) => {
+    console.log(`handling ${operation} response:`)
+    console.dir(response)
+    // HTTP status 409 == Conflict
+    if (operation == 'update' && response.status !== 409) {
+      const msg = 'The sinopia server is not accepting the request for this resource.'
+      this.state.updateResourceError ? this.setState({message: [`${msg}: ${this.state.updateResourceError}`]}) : this.setState({message: [msg]})
+    }
+
+    // HTTP status 401 == Unauthorized
```

mike sez this shouldn't be here - we should surface the error

---

done in my local branch